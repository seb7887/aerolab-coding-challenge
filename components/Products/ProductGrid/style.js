import styled from 'styled-components';

export const Grid = styled.div`
  padding: 1.5rem;
  display: grid;
  grid-template-columns: repeat(4, auto);
  grid-column-gap: 1.5rem;
  grid-row-gap: 2.5rem;

  @media (${props => props.theme.viewport.md}) {
    grid-template-columns: repeat(2, auto);
  }

  @media (${props => props.theme.viewport.sm}) {
    display: flex;
    flex-direction: column;
  }
`;
