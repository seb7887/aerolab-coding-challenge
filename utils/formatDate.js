export default function formatDate(value) {
  const getNumber = number => (number < 10 ? `0${number}` : String(number));
  const date = new Date(value);

  return `${getNumber(date.getMonth() + 1)}/${getNumber(
    date.getDate()
  )}/${getNumber(date.getFullYear())} ${getNumber(
    date.getUTCHours()
  )}:${getNumber(date.getUTCMinutes())}`;
}
