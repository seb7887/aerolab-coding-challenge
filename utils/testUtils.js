import faker from 'faker';

export const fakeUser = {
  name: 'John Kite',
  points: '7000'
};

export const mockCategories = [
  'all',
  'gaming',
  'laptops',
  'tablets',
  'cameras',
  'audio',
  'tv',
  'drones',
  'accessories',
  'home'
];

export const fakeProduct = {
  _id: 7,
  name: 'iPhone',
  category: 'Phones',
  cost: 800,
  img: {
    url: '/static/image/image.png'
  }
};

export const generateProduct = () => {
  const newProduct = {
    _id: faker.random.uuid(),
    name: faker.commerce.productName(),
    category: faker.commerce.productAdjective(),
    cost: 800,
    img: {
      url: faker.image.imageUrl()
    }
  };
  return newProduct;
};

export const fakeProducts = [fakeProduct];

export const generateProducts = quantity => {
  let products = [];

  for (let i = 0; i < quantity; i++) {
    const newProduct = generateProduct();
    products.push(newProduct);
  }

  return products;
};
