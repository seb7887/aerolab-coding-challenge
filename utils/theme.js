const black = '#000';
const white = '#fff';
const cyan = '#0ad4fa';
const orange = '#ff6e05';
const red = 'red';
const green = 'green';

const bgLight = '#e9ebf1';
const bgDark = '#212531';
const secondaryBgLigth = 'rgba(0, 0, 0, 0.1)';
const secondaryBgDark = 'rgba(255, 255, 255, 0.1)';
const containerBgLight = '#fff';
const containerBgDark = '#000';
const boxShadowLight = '0px 0px 3px 0px rgba(0, 0, 0, 0.1)';
const boxShadowDark = 'rgba(31, 31, 31, 0.1)';
const productShadowLight = '1px 4px 5px 1px rgba(0, 0, 0, 0.1)';
const productShadowDark = '1px 4px 5px 1px rgba(255, 255, 255, 0.1)';

const primaryLight = '#616161';
const primaryDark = '#f8f8f8';

const secondaryLight = black;
const secondaryDark = white;
const greyLight = '#e0e0e0';
const greyDark = '#a3a3a3';

const theme = {
  light: {
    black,
    white,
    cyan,
    orange,
    red,
    green,
    grey: greyLight,
    bg: bgLight,
    secondaryBg: secondaryBgLigth,
    containerBg: containerBgLight,
    boxShadow: boxShadowLight,
    productShadow: productShadowLight,
    primary: primaryLight,
    secondary: secondaryLight
  },
  dark: {
    black,
    white,
    cyan,
    orange,
    red,
    green,
    grey: greyDark,
    bg: bgDark,
    secondaryBg: secondaryBgDark,
    containerBg: containerBgDark,
    boxShadow: boxShadowDark,
    productShadow: productShadowDark,
    primary: primaryDark,
    secondary: secondaryDark
  }
};

export default theme;
