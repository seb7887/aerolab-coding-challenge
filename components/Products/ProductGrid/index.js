import React from 'react';

import Product from '../Product';

import { Grid } from './style';

const ProductGrid = ({ userPoints, products }) => (
  <Grid>
    {products.map(product => {
      return (
        <Product key={product._id} product={product} userPoints={userPoints} />
      );
    })}
  </Grid>
);

export default ProductGrid;
