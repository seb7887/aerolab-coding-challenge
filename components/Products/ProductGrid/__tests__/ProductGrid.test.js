import { render } from 'react-testing-library';

import ProductGrid from '../index';

import { generateProducts, fakeProducts } from '../../../../utils/testUtils';

const mockRedeemProduct = jest.fn();

const userPoints = 300;
const products = generateProducts(3);

describe('ProductGrid', () => {
  it('renders and matches snapshot', () => {
    const { asFragment } = render(
      <ProductGrid
        userPoints={userPoints}
        products={fakeProducts}
        redeemProduct={mockRedeemProduct}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should display a correct quantity of products', () => {
    const { getAllByTestId } = render(
      <ProductGrid
        userPoints={userPoints}
        products={products}
        redeemProduct={mockRedeemProduct}
      />
    );

    expect(getAllByTestId('product').length).toBe(products.length);
  });
});
