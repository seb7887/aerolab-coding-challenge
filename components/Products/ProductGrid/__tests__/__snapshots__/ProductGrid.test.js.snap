// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductGrid renders and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="style__Grid-sc-15ghthz-0 bNIUmK"
  >
    <div
      class="style__Container-g760h8-0 gPFdvq"
      data-testid="product"
    >
      <div
        class="style__Content-g760h8-5 tttFT"
      >
        <div
          class="buy"
        >
          <div
            class="style__NeedCoins-g760h8-1 aNIgB"
            data-testid="need-coins"
          >
            You need 500 
            <img
              alt="Coin"
              class="Coin__Img-sc-1i19xvt-0 cnqtJE"
              src="/static/images/coin.svg"
            />
          </div>
        </div>
        <img
          alt="iPhone"
          class="style__Img-g760h8-2 frYZGq"
          src="/static/image/image.png"
        />
        <div
          class="description"
        >
          <span
            class="category"
            data-testid="category"
          >
            Phones
          </span>
          <span
            class="name"
            data-testid="name"
          >
            iPhone
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
